src/pages/admin/team/components/delete-team-category-modal/DeleteTeamCategoryModal.tsx

```diff
@@ -0,0 +1,106 @@
+    const handleConfirmation = async () => {
+        if (isSubmitting || !selectedCategory || selectedCategory.teamMembersCount > 0) return;
+
+        try {
+            setError('');
+            setIsSubmitting(true);
+
+            await TeamCategoriesApi.delete(client, selectedCategory.id);
+
+            onConfirm(selectedCategory.id);
+            setSelectedCategory(undefined);
+            onClose();
+        } catch {
+            setError(COMMON_TEXT_ADMIN.CATEGORIES.FORM.MESSAGE.FAIL_TO_DELETE_CATEGORY);
+        } finally {
+            setIsSubmitting(false);
+        }
+    };
+
+    const handleClose = () => {
+        if (isSubmitting) return;
+        setSelectedCategory(undefined);
+        onClose();
+    };
+
+    useEffect(() => {
+        if (isOpen) {
+            setError('');
+            if (categories.length > 0) {
+                setSelectedCategory(categories[0]);
+            }
+        }
+    }, [isOpen, categories]);
+
+    const handleCategoryChange = (category: TeamCategory) => {
+        setSelectedCategory(category);
+    };
+
+    return (
+        <Modal isOpen={isOpen} onClose={handleClose}>
+            <Modal.Title>{COMMON_TEXT_ADMIN.CATEGORIES.FORM.TITLE.DELETE_CATEGORY}</Modal.Title>
+            <Modal.Content>
+                <div className="program-form-main">
+                    <SingleSelectInputGroup
+                        id="delete-category-select"
+                        label={COMMON_TEXT_ADMIN.CATEGORIES.FORM.LABEL.CATEGORY}
+                        isRequired={true}
+                        options={categories}
+                        getOptionId={(c) => c.id}
+                        getOptionName={(c) => c.name}
+                        placeholder=""
+                        onChange={handleCategoryChange}
+                        disabled={isSubmitting}
+                        value={selectedCategory}
+                    />
+                    {selectedCategory && selectedCategory.teamMembersCount > 0 && (
+                        <HintBox
+                            title={TEAM_CATEGORY_VALIDATION.teamMembersCount.getHasTeamMembersCountError(
+                                selectedCategory.teamMembersCount,
+                            )}
+                            text={TEAM_CATEGORY_VALIDATION.teamMembersCount.getRelocationOrRemovalHint()}
+                        />
+                    )}
+                    {error && <div className="team-category-modal-error-container">{error}</div>}
+                </div>
+            </Modal.Content>
+            <Modal.Actions>
+                <Button buttonStyle="secondary" onClick={handleClose} disabled={isSubmitting}>
+                    {COMMON_TEXT_ADMIN.BUTTON.CANCEL}
+                </Button>
+                <Button
+                    buttonStyle="primary"
+                    onClick={handleConfirmation}
+                    disabled={(!!selectedCategory && selectedCategory.teamMembersCount > 0) || isSubmitting}
```

Comment:
Use derived state for complex disabled logic
```tsx
const isCategoryDeletable = selectedCategory && selectedCategory.teamMembersCount === 0;
const isDeleteDisabled = !isCategoryDeletable || isSubmitting;
```
```suggestion
                    disabled={isDeleteDisabled}
```

Reply:
Done